dropwizard-hibernate/src/test/java/io/dropwizard/hibernate/DataExceptionMapper.java

```diff
@@ -0,0 +1,26 @@
+package io.dropwizard.hibernate;
+
+import com.google.common.collect.ImmutableMap;
+import org.hibernate.exception.DataException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+@Provider
+public class DataExceptionMapper implements ExceptionMapper<DataException> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataException.class);
+
+    @Override
+    public Response toResponse(DataException e) {
+        LOGGER.error("Hibernate error", e);
+        String message = e.getCause().getMessage().contains("EMAIL") ? "Wrong email" : "Wrong input";
+
+        return Response.status(Response.Status.BAD_REQUEST)
+                .entity(ImmutableMap.of("error", message))
```

Comment:
That makes sense. Reused `ErrorMessage` for error message generation and testing.